code/datums/components/living_no_move.dm

```diff
@@ -0,0 +1,19 @@
+/datum/component/living_no_move
+
+// Will make a living thing unable to move
+/datum/component/living_no_move/Initialize(...)
+	. = ..()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/mob/living/M = parent
+	M.canmove = FALSE
+	RegisterSignal(parent, COMSIG_LIVING_UPDATE_CAN_MOVE, .proc/update_canmove)
+
+/datum/component/living_no_move/UnregisterFromParent()
+	var/mob/living/M = parent
+	M.canmove = TRUE
+	UnregisterSignal(parent, COMSIG_LIVING_UPDATE_CAN_MOVE)
+
+/datum/component/living_no_move/proc/update_canmove()
+	var/mob/living/M = parent
+	M.canmove = FALSE
```

Comment:
I really don't get the point of this, to be honest, when there's multiple effects that can be applied that prevent someone from moving---like `Stun`. 

I also don't like procs that have the same exact proc name as a different proc, but their functionality is different (but *semi* related). It's bad practice and just generates confusion.

Reply:
Will change the naming.
Some context why I made it.
I wanted to have abilities take a long time and once they were channeling I wanted the user to be unable to move. Using stun here would break the channel since I use a do_after